test/e2e/tests/test_open_id_connect_provider.py

```diff
@@ -0,0 +1,176 @@
+
+        condition.assert_synced(ref)
+
+        logging.debug(f"\n\n**** OIDCProvider create validation")
+        latest_oidcp_boto3 = open_id_connect_provider.get(oidc_provider_arn)
+        logging.debug(f"\n**** OIDCProvider created: {latest_oidcp_boto3}")
+
+        assert latest_oidcp_boto3 is not None
+        assert len(latest_oidcp_boto3["ThumbprintList"]) == 1
+        latest_oidcp_url = latest_oidcp_boto3["Url"]
+        assert_url_equals_ignore_prefix(latest_oidcp_url, "https://host.domain.net")
+
+        # perform an update to some part of the OIDCProvider
+        logging.debug(f"\n\n**** OIDCProvider update")
+
+        new_thumbprints = [
+            "9876543210987654321098765432109876543210"
+        ]  # thumbprints must be 40 characters
+        updates = {
+            "spec": {
+                "thumbprintList": new_thumbprints,
+                "tags": [{"key": "key2", "value": "val2"}],
+            },
+        }
+        k8s.patch_custom_resource(ref, updates)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+        logging.debug(f"\n\n**** OIDCProvider update validation")
+        latest_oidcp_boto3 = open_id_connect_provider.get(oidc_provider_arn)
+        assert latest_oidcp_boto3 is not None
+        logging.debug(f"\n\n**** OIDCProvider updated: {latest_oidcp_boto3}")
+        assert len(latest_oidcp_boto3["ThumbprintList"]) == 1
+        assert latest_oidcp_boto3["ThumbprintList"][0] == new_thumbprints[0]
+
+        after_update_expected_tags = [{"Key": "key2", "Value": "val2"}]
+        logging.debug(f"\n\n**** OIDCProvider validate tags update")
+        latest_tags = open_id_connect_provider.get_tags(oidc_provider_arn)
+        assert tag.cleaned(latest_tags) == after_update_expected_tags
+
+        # validate that changing the URL results in a terminal condition
+        update_url = {
+            "spec": {
+                "url" : "https://some.other.domain.com"
+            }
+        }
+        logging.debug(f"\n\n**** OIDCProvider update of URL intended to fail")
+        k8s.patch_custom_resource(ref, update_url)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+        cr = k8s.get_resource(ref)
+        logging.debug(f"\n\n**** OIDCProvider CR updated: {cr}")
+        assert cr is not None
+        assert "status" in cr
+        cr_conditions = cr["status"]["conditions"]
+        cond_synced_found = False
+        cond_terminal_found = False
+        for cond in cr_conditions:
+            if cond["type"] == "ACK.ResourceSynced":
+                assert cond["status"] == False or cond["status"] == "False"
+                cond_synced_found = True
+            if cond["type"] == "ACK.Terminal":
+                assert cond["status"] == True or cond["status"] == "True"
+                assert cond["message"] == "Immutable Spec fields have been modified: URL"
+                cond_terminal_found = True
+        assert cond_synced_found and cond_terminal_found
```

Comment:
I think the `condition` helpers might be useful for you here: https://github.com/aws-controllers-k8s/test-infra/blob/main/src/acktest/k8s/condition.py

If not, we should add these to the common testing package

Reply:
Done